core/shared/platform/windows/win_socket.c

```diff
@@ -11,6 +11,19 @@
 
 static bool is_winsock_inited = false;
 
+#define CHECK_VALID_SOCKET_HANDLE(win_handle)                   \
+    do {                                                        \
+        if ((win_handle) == NULL                                \
+            || (win_handle)->raw.socket == INVALID_SOCKET) {    \
+            errno = EBADF;                                      \
+            return BHT_ERROR;                                   \
+        }                                                       \
+        if ((win_handle)->type != windows_handle_type_socket) { \
```

Comment:
I'd probably do this check before checking for `INVALID_SOCKET`. I think in the context of the file it's more descriptive to know that the type is invalid than the socket is invalid. Also, checking for the value of the union before checking its type seems logically incorrect (even though in this specific case it might work).